[WIP] Pocket Network Implementation

studna opened the pull request:
# Motivation

I would love to add POKT support.

## About Pocket Network

[Pocket Network](https://www.pokt.network/) is a decentralized node infrastructure protocol that enables users (applications/developers) to access blockchain data. Users submit API requests ([relay requests](https://docs.pokt.network/learn/protocol/servicing/#overview)) to service nodes via access points ([Gateways](https://www.pokt.network/one-small-step-for-nodies-one-giant-leap-for-pokt-network/)) that build on top of the POKT Network. [source](https://messari.io/project/pocket-network-2/quarterly-reports/q4-2023)

## Changes Made

- Implemented the necessary functions to interact with Pocket Network.
- Updated the user interface to include options for Pocket Network.
- Installed two external NPM packages `protobufjs` and `long`

## Limitations

Currently, the CI/CD pipeline for Kraken Wallet is hosted on GitLab (just guessing) and is not open-sourced, making contributions challenging for open-source developers. It would be great to open-source development tools, instructions, the tests directory, and lockfiles.

## Todos

- [ ] can Harmony API support POKT or do we need to use Pocket RPC Endpoint?
- [ ] properly handle `fetchTransactions` - i drafted the implementation which is working, but it renders tx as "contract interaction"
- [ ] create and use `/assets/images/assetGroup/asset-group-pokt.png` for empty state (currently it's using default image)
- [ ] use some API to obtain USD value of POKT holdings

## Disclaimer

I'm not affiliated with Pocket Network.

## Screenshots

![image](https://github.com/krakenfx/wallet/assets/939784/85d37ca0-e52b-407b-8e01-a3c65a2235fe)

![image](https://github.com/krakenfx/wallet/assets/939784/c183f761-71e5-4e44-b1e1-a0e9755feef9)

![Screenshot 2024-04-20 at 0 33 37](https://github.com/krakenfx/wallet/assets/939784/f6ebcc4a-2fc5-430c-ab70-2d4bf8583322)

Transaction that was sent by Kraken Wallet (using iOS Emulator): https://poktscan.com/tx/99BA508138ED6205DDE9A531995100978284E5ADE94220496F67B7617AA756A1

erickuhn19 commented:
@studna is the point of this PR to switch from using the default nodes Kraken Wallet supports to using a node run by the POKT network? Can you give more background on the purpose?

studna commented:
@erickuhn19 The scope of this PR is limited to adding POKT as another wallet/asset supported by Kraken Wallet, essentially allowing users to send and receive POKT on the Pocket Network blockchain (as you can see on screenshots).

I'm not sure if the Harmony API can support the POKT blockchain (or when it might be able to), so I implemented a pocket-specific transport that uses the Pocket mainnet RPC Endpoint provided by Grove.city (the official POKT Gateway). Based on this implementation it can be even implemented under Harmony API and replaced with Harmony Transport. 

> is the point of this PR to switch from using the default nodes Kraken Wallet supports to using a node run by the POKT network?

While the ability to switch from the Harmony API to decentralized RPCs for all networks would be a positive step toward a more transparent and decentralized wallet, that's not the purpose of this PR.

studna commented:
@erickuhn19 I’m closing this PR as it’s becoming outdated. Pocket Network is transitioning to [a native ERC20](https://github.com/MetaMask/contract-metadata/pull/1336), and while this implementation works, it will be deprecated with a protocol upgrade by the end of the year.
